lib/note-detail.jsx

```diff
@@ -34,22 +30,34 @@ export default React.createClass( {
 		return note && note.id;
 	},
 
-	componentWillReceiveProps: function( { note } ) {
-		this.setState( {
-			content: get( note, 'data.content', '' )
-		} );
+	componentWillReceiveProps: function() {
+		const { note } = this.props;
+
+		if ( ! this.isValidNote( note ) || ! this.noteEditor ) return;
+
+		if ( saveTimerId ) clearTimeout( saveTimerId );
+
+		// Save note if content changes haven't been saved yet
+		if ( note.data.content !== this.noteEditor.value ) {
+			this.props.onChangeContent( note, this.noteEditor.value );
+			analytics.tracks.recordEvent( 'editor_note_edited' );
+		}
```

Comment:
Is the purpose of this function to make sure that an incoming update from the server doesn't overwrite the content that we have already changed but not yet dispatched? In other words, if we have stopped typing for two seconds and we get an update from the server, is this code supposed to prevent that update from wiping out what I did two second ago?